warehouse/query-core/src/test/java/datawave/query/jexl/JexlNodeFactoryTest.java

```diff
@@ -0,0 +1,179 @@
+package datawave.query.jexl;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import datawave.query.jexl.functions.arguments.JexlArgument;
+import datawave.query.jexl.lookups.IndexLookupMap;
+import datawave.query.jexl.lookups.ValueSet;
+import datawave.query.jexl.visitors.JexlStringBuildingVisitor;
+import org.apache.commons.jexl2.parser.ASTEQNode;
+import org.apache.commons.jexl2.parser.ASTERNode;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.JexlNode;
+import org.apache.commons.jexl2.parser.ParseException;
+import org.apache.commons.jexl2.parser.ParserTreeConstants;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class JexlNodeFactoryTest {
+    
+    @Test
+    public void testNoFieldExpansion() throws ParseException {
+        IndexLookupMap indexLookupMap = new IndexLookupMap(10, 10);
+        indexLookupMap.putAll("FOO", Sets.newHashSet("6", "9"));
+        indexLookupMap.putAll("BAR", Sets.newHashSet("7", "8"));
+        
+        JexlNode originalNode = JexlASTHelper.parseJexlQuery("((_Bounded_ = true) && (FOO > 5 && FOO < 10))");
+        JexlNode node = JexlNodeFactory.createNodeTreeFromFieldsToValues(JexlNodeFactory.ContainerType.OR_NODE, false, originalNode, indexLookupMap, false,
+                        true, false);
+        
+        assertEquals("(FOO == '6' || FOO == '7' || FOO == '8' || FOO == '9')", JexlStringBuildingVisitor.buildQuery(node));
+        assertTrue(JexlASTHelper.validateLineage(node, false));
```

Comment:
should invalid lineage fail hard here and in other new tests?

Reply:
Failing hard throws an exception, validating a lineage with failHard=false let's us say "this assertion failed" not "JexlASTHelper threw an exception"